adobe_vipm/flows/validation/transfer.py

```diff
@@ -304,205 +314,195 @@ def validate_transfer_not_migrated(mpt_client, adobe_client, order):
-
-    Args:
-        subscriptions: The subscriptions to update
-        adobe_transfer: The Adobe transfer data
-
-    Returns:
-        list: Updated subscriptions
-    """
-    for subscription in subscriptions["items"]:
-        correct_sku = get_transfer_item_sku_by_subscription(
-            adobe_transfer,
-            subscription["subscriptionId"]
+        context.customer = customer
+
+        if len(context.subscriptions["items"]) == 0:
+            if customer.get("globalSalesEnabled", False):
+                logger.error(ERR_NO_SUBSCRIPTIONS_WITHOUT_DEPLOYMENT)
+                param = get_ordering_parameter(context.order, PARAM_MEMBERSHIP_ID)
+                context.order = set_ordering_parameter_error(
+                    context.order,
+                    PARAM_MEMBERSHIP_ID,
+                    ERR_ADOBE_MEMBERSHIP_ID.to_dict(
+                        title=param["name"],
+                        details=ERR_NO_SUBSCRIPTIONS_WITHOUT_DEPLOYMENT
+                    ),
+                )
+                context.validation_succeeded = False
+                return
+            context.validation_succeeded = True
+            return
+
+        next_step(mpt_client, context)
+
+class AddLinesToOrder(Step):
+    def __call__(self, mpt_client, context, next_step):
+        if not context.transfer:
+            has_error, order = validate_transfer_not_migrated(
+                mpt_client, context.order
+            )
+            context.order = order
+            context.validation_succeeded = not has_error
+            return
+
+        commitment = get_3yc_commitment(context.customer)
+        has_error, order = add_lines_to_order(
+            mpt_client,
+            context.order,
+            context.subscriptions["items"],
+            commitment,
+            "currentQuantity",
+            True
         )
-        subscription["offerId"] = correct_sku or subscription["offerId"]
-    return subscriptions
+        context.order = order
+        context.validation_succeeded = not has_error
+        next_step(mpt_client, context)
 
-def _validate_empty_subscriptions(subscriptions, customer, order):
+def validate_transfer(mpt_client, order):
     """
-    Validates if there are no subscriptions to transfer.
-
-    Args:
-        subscriptions: The subscriptions to validate
-        customer: The customer data
-        order: The order being validated
-
-    Returns:
-        tuple: (has_error, order) where has_error is True if validation failed
+    Valida una orden de transferencia usando un pipeline de pasos.
     """
-    if len(subscriptions["items"]) == 0:
-        if customer.get("globalSalesEnabled", False):
-            logger.error(ERR_NO_SUBSCRIPTIONS_WITHOUT_DEPLOYMENT)
-            return _set_membership_error(
-                order,
-                ERR_NO_SUBSCRIPTIONS_WITHOUT_DEPLOYMENT
-            )
-        return False, order
-    return None, order
+    pipeline = Pipeline(
```

Comment:
I would say this split to steps has several problems:

1. Context is overused :-( There are a lot of things used there. I would try to minify number of variables shared in context
2. Ordering is really strange :-( validating inactive account after subscriptions are fetched or validate empty subscriptions after subscriptions skus are updated :-(

I would make steps less coupled and more logical.

1. SetupTransferContext() - it should just fulfill order and that's it, most probably original SetupContext can be reused
2. FetchTransfer() - fetches transfer information
3. ValidateTransferStatus() - validates transfer status (all statuses, for all transfer)
4. ....